yasp.c

```diff
@@ -68,30 +66,27 @@ int16_t process_yasp(uint8_t *buffer, uint16_t length)
     if (i == registered_commands) {
         return RET_CMD_NOT_REGISTERED;
     }
-    return (START_CHECKSUM + cmd_len + 1);
+    return (START_CRC32 + cmd_len + 1);
 }
 
 void send_yasp_ack(uint8_t cmd, uint8_t * payload, uint16_t length)
 {
     int i;
     uint8_t out_buffer[2+2+1];
-    uint8_t checksum = 0x00;
+    uint32_t crc_32 = 0x00;
     out_buffer[SYNCH1_POS] = 0xFF;
     out_buffer[SYNCH2_POS] = 0xFF;
     out_buffer[LENGTH_POS] = (uint8_t)((length + 3) >> 8);
     out_buffer[LENGTH_POS+1] = (uint8_t)(length+3);
     out_buffer[COMMAND_POS] = 0x80 | cmd;
     serial_tx(out_buffer, 5);
-    checksum += out_buffer[2] + out_buffer[3] + out_buffer[4];
+    crc_32 = (crc_32, &out_buffer[2], 3);
```

Comment:
Same here, the tests probably pass because you have the same mistake in two places.  Seems like we need a failing test case.